src/aggregator/aggregator/entry.go

```diff
@@ -925,28 +902,35 @@ func (e *Entry) addTimedWithLock(
 }
 
 func (e *Entry) addTimedWithStagedMetadatasAndLock(metric aggregated.Metric) error {
-	timestamp := time.Unix(0, metric.TimeNanos)
-	multiErr := xerrors.NewMultiError()
+	var (
+		timestamp = time.Unix(0, metric.TimeNanos)
+		err       error
+	)
+
 	for i := range e.aggregations {
-		err := e.checkTimestampForTimedMetric(metric, e.nowFn().UnixNano(),
-			e.aggregations[i].key.storagePolicy.Resolution().Window)
-		if err != nil {
-			multiErr = multiErr.Add(err)
+		if multierr.AppendInto(
```

Comment:
FYI @vdarulis this broke the error handling when there are multiple errors :(

https://github.com/m3db/m3/pull/4000